In [1]:
import pandas as pd

In [2]:
cols = ['Nullzero','QueryTime','QuerySeq','AvatarID','Guild','Level','Race','Class','Zone','Ignore1','Ignore2']

In [3]:
df = pd.read_csv("H:\Output\wowlogs.csv", names = cols, usecols = ['QueryTime','QuerySeq','AvatarID','Guild','Level','Race','Class','Zone'])

In [4]:
df.head(5)

QueryTime  QuerySeq  AvatarID Guild  Level  Race     Class  \
0   12/31/05 23:59:46         1         0            5   Orc   Warrior   
1   12/31/05 23:59:46         1         1            9   Orc    Shaman   
2   12/31/05 23:59:52         2         2           13   Orc    Shaman   
3   12/31/05 23:59:52         2         3     0     14   Orc   Warrior   
4   12/31/05 23:59:52         2         4           14   Orc    Shaman   

       Zone  
0   Durotar  
1   Durotar  
2   Durotar  
3   Durotar  
4   Durotar

In [5]:
df.tail(5)

QueryTime  QuerySeq  AvatarID Guild  Level        Race  \
41063250   01/10/09 05:08:48        56     36893   104     80   Blood Elf   
41063251   01/10/09 05:08:48        56     39532   204     80   Blood Elf   
41063252   01/10/09 05:08:59        58     90033   502     80   Blood Elf   
41063253   01/10/09 05:08:59        58     87974   251     80   Blood Elf   
41063254   01/10/09 05:08:59        58     86679   459     80   Blood Elf   

                  Class                     Zone  
41063250           Mage                  Dalaran  
41063251           Mage          The Storm Peaks  
41063252   Death Knight           Sholazar Basin  
41063253   Death Knight   Blade's Edge Mountains  
41063254   Death Knight        Shadowmoon Valley

In [6]:
total_records = df.count()

<html>
<font color = DarkBlue>
List all unique races from data set.
</font>
</html>

In [7]:
df.Race.unique()

array([' Orc', ' Tauren', ' Troll', ' Undead', '373族', '547人', '3033',
       '27410', ' Blood Elf', '74622妖'], dtype=object)

<html>
<font color = DarkBlue>
Following values are incorrect Warcraft races: <br />
'373族', '547人', '3033','27410', '74622妖' <br />
Let us look at the records which have these incorrect races.
</font>
</html>

In [8]:
df_incorrect_race = df[df['Race'].isin(['373族', '547人', '3033','27410', '74622妖'])]

In [9]:
df_incorrect_race.AvatarID.unique()

array([  373,   547,  3033, 27410, 74622], dtype=int64)

In [10]:
df_incorrect_race.count()

QueryTime    50085
QuerySeq     50085
AvatarID     50085
Guild        50085
Level        50085
Race         50085
Class        50085
Zone         50085
dtype: int64

<html>
<font color = DarkBlue>
50085 records that belongs to 5 users have incorrect race captured in the logs. These records need to be removed to avoid incorrect analysis.
</font>
</html>

<html>
<font color = DarkBlue>
List all unique classes from data set.
</font>
</html>

In [11]:
df.Class.unique()

array([' Warrior', ' Shaman', ' Hunter', ' Warlock', ' Rogue', ' Druid',
       ' Mage', ' Priest', '482', '2400', '3485伊', ' Paladin',
       ' Death Knight'], dtype=object)

<html>
<font color = DarkBlue>
Following values are incorrect Warcraft classes: <br />
'482', '2400', '3485伊' <br />
Let us look at the records which have these incorrect classes.
</font>
</html>

In [12]:
df_incorrect_class = df[df['Class'].isin(['482', '2400', '3485伊'])]

In [13]:
df_incorrect_class.AvatarID.unique()

array([ 482, 2400, 3485], dtype=int64)

In [14]:
df_incorrect_class.count()

QueryTime    376
QuerySeq     376
AvatarID     376
Guild        376
Level        376
Race         376
Class        376
Zone         376
dtype: int64

<html>
<font color = DarkBlue>
376 records that belongs to 3 users have incorrect class captured in the logs. These records need to be removed to avoid incorrect analysis.
</font>
</html>

<html>
<font color = DarkBlue>
List all unique zones from data set.
</font>
</html>

In [15]:
df.Zone.nunique()

173

In [16]:
df.Zone.unique()

array([' Durotar', ' The Barrens', ' Silverpine Forest',
       ' Stonetalon Mountains', ' Thunder Bluff', ' Dustwallow Marsh',
       ' Orgrimmar', ' Undercity', ' Ashenvale', ' Stranglethorn Vale',
       ' Wailing Caverns', ' Tanaris', ' Maraudon', ' The Hinterlands',
       " Un'Goro Crater", ' Felwood', " The Temple of Atal'Hakkar",
       ' Blackrock Depths', ' Winterspring', ' Eastern Plaguelands',
       ' Azshara', ' Scholomance', ' Arathi Basin', ' Blackrock Spire',
       ' Blackrock Mountain', ' Moonglade', ' Mulgore',
       ' Thousand Needles', ' Hillsbrad Foothills', ' Blackfathom Deeps',
       ' Desolace', ' Darkshore', ' Alterac Mountains',
       ' Swamp of Sorrows', ' Tirisfal Glades', ' Searing Gorge',
       ' Burning Steppes', " Zul'Farrak", ' Feralas', ' Arathi Highlands',
       ' Blasted Lands', ' Western Plaguelands', ' Stratholme',
       ' Scarlet Monastery', ' Shadowfang Keep', ' Badlands', ' 未知',
       ' Wetlands', ' Redridge Mountains', ' Hall of Legend

<html>
<font color = DarkBlue>
Let us look at the records which have these incorrect warcraft Zones captured in their logs.
</font>
</html>

In [17]:
df_incorrect_zone = df[~df['Zone'].isin([' Durotar', ' The Barrens', ' Silverpine Forest',
       ' Stonetalon Mountains', ' Thunder Bluff', ' Dustwallow Marsh',
       ' Orgrimmar', ' Undercity', ' Ashenvale', ' Stranglethorn Vale',
       ' Wailing Caverns', ' Tanaris', ' Maraudon', ' The Hinterlands',
       " Un'Goro Crater", ' Felwood', " The Temple of Atal'Hakkar",
       ' Blackrock Depths', ' Winterspring', ' Eastern Plaguelands',
       ' Azshara', ' Scholomance', ' Arathi Basin', ' Blackrock Spire',
       ' Blackrock Mountain', ' Moonglade', ' Mulgore',
       ' Thousand Needles', ' Hillsbrad Foothills', ' Blackfathom Deeps',
       ' Desolace', ' Darkshore', ' Alterac Mountains',
       ' Swamp of Sorrows', ' Tirisfal Glades', ' Searing Gorge',
       ' Burning Steppes', " Zul'Farrak", ' Feralas', ' Arathi Highlands',
       ' Blasted Lands', ' Western Plaguelands', ' Stratholme',
       ' Scarlet Monastery', ' Shadowfang Keep', ' Badlands',
       ' Wetlands', ' Redridge Mountains',
       ' Hall of Legends', ' Gnomeregan', ' Razorfen Downs', ' Duskwood',
       ' Loch Modan', ' Silithus', ' Deadwind Pass', ' Razorfen Kraul',
       ' Ragefire Chasm', ' Dire Maul', ' Uldaman', ' Dun Morogh',
       ' Elwynn Forest', " Zul'Gurub", ' Teldrassil', ' Westfall',
       ' Stormwind City', ' Molten Core', " Onyxia's Lair",
       ' Warsong Gulch', ' Blackwing Lair', ' Deadmines',
       " Gates of Ahn'Qiraj", 
       ' Alterac Valley',
       ' Ironforge',
       ' Hyjal',
       ' Deeprun Tram', " Ruins of Ahn'Qiraj", " Ahn'Qiraj", ' Naxxramas',
       ' GM Island', ' The Great Sea', ' Nagrand Arena',
       " Blade's Edge Arena", " Quel'thalas", ' Hellfire Peninsula',
       ' Ghostlands', ' Silvermoon City', ' Zangarmarsh',
       ' Terokkar Forest', ' Shattrath City', ' Hellfire Ramparts',
       ' Eversong Woods', ' Nagrand', " Blade's Edge Mountains",
       ' Netherstorm', ' The Blood Furnace', ' Shadowmoon Valley',
       ' Coilfang: The Slave Pens', ' The Steamvault', ' Eye of the Storm',
       ' Azuremyst Isle', ' Bloodmyst Isle', ' Mana-Tombs',
       ' Old Hillsbrad Foothills', ' Auchenai Crypts', ' Sethekk Halls',
       ' Twisting Nether', ' The Black Morass', ' The Veiled Sea',
       ' The Shattered Halls', ' Auchindoun: Shadow Labyrinth',
       ' The Botanica', ' The Arcatraz', " Gruul's Lair", ' Karazhan',
       " Magtheridon's Lair", ' The Exodar', ' The Forbidding Sea',
       ' Ruins of Lordaeron', ' Serpentshrine Cavern', ' Tempest Keep',
       ' The Underbog', ' The Mechanar',
       ' Black Temple', " Magisters' Terrace", " Isle of Quel'Danas",
       ' Sunwell Plateau', ' The North Sea', ' The Ring of Valor',
       ' Plaguelands: The Scarlet Enclave', ' Borean Tundra',
       ' Dragonblight', ' Howling Fjord', ' The Nexus', ' Grizzly Hills',
       ' Sholazar Basin', ' Crystalsong Forest', ' The Frozen Sea',
       ' Utgarde Keep', ' Azjol-Nerub', ' The Storm Peaks', ' Icecrown',
       ' Dalaran', " Ahn'kahet: The Old Kingdom", " Zul'Drak",
       ' The Violet Hold', ' Wintergrasp', ' Strand of the Ancients',
       " Drak'Tharon Keep", ' Gundrak', ' The Culling of Stratholme',
       ' The Oculus', ' Halls of Lightning', ' Halls of Stone',
       ' Utgarde Pinnacle', ' Vault of Archavon', ' The Eye of Eternity',
       ' The Obsidian Sanctum'])]

In [18]:
df_incorrect_zone.Zone.unique()

array([' 未知', ' 監獄', ' 時光洞穴', ' 達納蘇斯', '8585', '1608峽谷', '2029', '15641',
       '1007城', ' 北方海岸', ' 毒牙沼澤', ' 麥克那爾', '61477', ' 龍骨荒野', '1231崔茲',
       ' Dalaran競技場'], dtype=object)

In [19]:
df_incorrect_zone.count()

QueryTime    370298
QuerySeq     370298
AvatarID     370298
Guild        370298
Level        370298
Race         370298
Class        370298
Zone         370298
dtype: int64

In [20]:
df_incorrect_zone.AvatarID.nunique()

5441

<html>
<font color = DarkBlue>
370298 records that belongs to 5441 users have incorrect Zone captured in the logs. These records need to be removed to avoid incorrect analysis.
</font>
</html>

<html>
<font color = DarkBlue>
Now, Let us clean up data by removing all incorrect records of Race, Class and Zone.
</font>
</html>

In [21]:
df_final = df[df['Race'].isin([' Orc', ' Tauren', ' Troll', ' Undead',' Blood Elf'])]

In [22]:
df_final = df_final[df_final['Class'].isin([' Warrior', ' Shaman', ' Hunter', ' Warlock', ' Rogue', ' Druid',
       ' Mage', ' Priest',' Paladin', ' Death Knight'])]

In [23]:
df_final = df_final[df_final['Zone'].isin([' Durotar', ' The Barrens', ' Silverpine Forest',
       ' Stonetalon Mountains', ' Thunder Bluff', ' Dustwallow Marsh',
       ' Orgrimmar', ' Undercity', ' Ashenvale', ' Stranglethorn Vale',
       ' Wailing Caverns', ' Tanaris', ' Maraudon', ' The Hinterlands',
       " Un'Goro Crater", ' Felwood', " The Temple of Atal'Hakkar",
       ' Blackrock Depths', ' Winterspring', ' Eastern Plaguelands',
       ' Azshara', ' Scholomance', ' Arathi Basin', ' Blackrock Spire',
       ' Blackrock Mountain', ' Moonglade', ' Mulgore',
       ' Thousand Needles', ' Hillsbrad Foothills', ' Blackfathom Deeps',
       ' Desolace', ' Darkshore', ' Alterac Mountains',
       ' Swamp of Sorrows', ' Tirisfal Glades', ' Searing Gorge',
       ' Burning Steppes', " Zul'Farrak", ' Feralas', ' Arathi Highlands',
       ' Blasted Lands', ' Western Plaguelands', ' Stratholme',
       ' Scarlet Monastery', ' Shadowfang Keep', ' Badlands',
       ' Wetlands', ' Redridge Mountains',
       ' Hall of Legends', ' Gnomeregan', ' Razorfen Downs', ' Duskwood',
       ' Loch Modan', ' Silithus', ' Deadwind Pass', ' Razorfen Kraul',
       ' Ragefire Chasm', ' Dire Maul', ' Uldaman', ' Dun Morogh',
       ' Elwynn Forest', " Zul'Gurub", ' Teldrassil', ' Westfall',
       ' Stormwind City', ' Molten Core', " Onyxia's Lair",
       ' Warsong Gulch', ' Blackwing Lair', ' Deadmines',
       " Gates of Ahn'Qiraj", 
       ' Alterac Valley',
       ' Ironforge',
       ' Hyjal',
       ' Deeprun Tram', " Ruins of Ahn'Qiraj", " Ahn'Qiraj", ' Naxxramas',
       ' GM Island', ' The Great Sea', ' Nagrand Arena',
       " Blade's Edge Arena", " Quel'thalas", ' Hellfire Peninsula',
       ' Ghostlands', ' Silvermoon City', ' Zangarmarsh',
       ' Terokkar Forest', ' Shattrath City', ' Hellfire Ramparts',
       ' Eversong Woods', ' Nagrand', " Blade's Edge Mountains",
       ' Netherstorm', ' The Blood Furnace', ' Shadowmoon Valley',
       ' Coilfang: The Slave Pens', ' The Steamvault', ' Eye of the Storm',
       ' Azuremyst Isle', ' Bloodmyst Isle', ' Mana-Tombs',
       ' Old Hillsbrad Foothills', ' Auchenai Crypts', ' Sethekk Halls',
       ' Twisting Nether', ' The Black Morass', ' The Veiled Sea',
       ' The Shattered Halls', ' Auchindoun: Shadow Labyrinth',
       ' The Botanica', ' The Arcatraz', " Gruul's Lair", ' Karazhan',
       " Magtheridon's Lair", ' The Exodar', ' The Forbidding Sea',
       ' Ruins of Lordaeron', ' Serpentshrine Cavern', ' Tempest Keep',
       ' The Underbog', ' The Mechanar',
       ' Black Temple', " Magisters' Terrace", " Isle of Quel'Danas",
       ' Sunwell Plateau', ' The North Sea', ' The Ring of Valor',
       ' Plaguelands: The Scarlet Enclave', ' Borean Tundra',
       ' Dragonblight', ' Howling Fjord', ' The Nexus', ' Grizzly Hills',
       ' Sholazar Basin', ' Crystalsong Forest', ' The Frozen Sea',
       ' Utgarde Keep', ' Azjol-Nerub', ' The Storm Peaks', ' Icecrown',
       ' Dalaran', " Ahn'kahet: The Old Kingdom", " Zul'Drak",
       ' The Violet Hold', ' Wintergrasp', ' Strand of the Ancients',
       " Drak'Tharon Keep", ' Gundrak', ' The Culling of Stratholme',
       ' The Oculus', ' Halls of Lightning', ' Halls of Stone',
       ' Utgarde Pinnacle', ' Vault of Archavon', ' The Eye of Eternity',
       ' The Obsidian Sanctum'])]

In [24]:
final_records = df_final.count()

In [25]:
removed_records = total_records - final_records

In [26]:
total_records, final_records , removed_records

(QueryTime    41063255
 QuerySeq     41063255
 AvatarID     41063255
 Guild        41063255
 Level        41063255
 Race         41063255
 Class        41063255
 Zone         41063255
 dtype: int64, QueryTime    40642764
 QuerySeq     40642764
 AvatarID     40642764
 Guild        40642764
 Level        40642764
 Race         40642764
 Class        40642764
 Zone         40642764
 dtype: int64, QueryTime    420491
 QuerySeq     420491
 AvatarID     420491
 Guild        420491
 Level        420491
 Race         420491
 Class        420491
 Zone         420491
 dtype: int64)

In [27]:
(removed_records/total_records)*100

QueryTime    1.024008
QuerySeq     1.024008
AvatarID     1.024008
Guild        1.024008
Level        1.024008
Race         1.024008
Class        1.024008
Zone         1.024008
dtype: float64

<html>
<font color = DarkBlue>
We will have to remove 1% of the records (420491 out of 41063255) to avoid incorrect analysis and inferences from warcraft logs.
This data is relatively less compared to the total size of warcraft logs we have gathered.
</font>
</html>

<html>
<font color = DarkBlue>
Save the final set of records into a new csv file to be used in further steps.
</font>
</html>

In [28]:
df_final.to_csv("newlogs.csv", sep=',', encoding='utf-8')

In [3]:
df_final = pd.read_csv("newlogs.csv", usecols = ['QueryTime','QuerySeq','AvatarID','Guild','Level','Race','Class','Zone'])